ocean_lib/models/v4/test/test_factory_router.py

```diff
@@ -0,0 +1,444 @@
+    ocean_tokens = factory_router.ocean_tokens(get_ocean_address(config))
+    assert ocean_tokens is True
+
+
+def test_add_token(web3, config):
+    newTokenAddress = ERC20Token.deploy(
+        web3=web3, deployer_wallet=get_factory_deployer_wallet(network=_NETWORK)
+    )
+    factory_router = FactoryRouter(web3, get_factory_router_address(config))
+
+    print(get_factory_deployer_wallet(network=_NETWORK).address)
+    print(factory_router.router_owner())
+
+    assert factory_router.ocean_tokens(newTokenAddress) == False
+    factory_router.add_ocean_token(
+        newTokenAddress, get_factory_deployer_wallet(network=_NETWORK)
+    )
+    assert factory_router.ocean_tokens(newTokenAddress) == True
+
+
+def test_fail_add_token(web3, config):
+    newTokenAddress = get_erc_address(config)
+    factory_router = FactoryRouter(web3, get_factory_router_address(config))
+    with pytest.raises(exceptions.ContractLogicError):
+        factory_router.add_ocean_token(newTokenAddress, get_another_consumer_wallet())
+
+
+def test_remove_token(web3, config):
+    newTokenAddress = create_new_token(web3, config)
+
+    factory_router = FactoryRouter(web3, get_factory_router_address(config))
+    factory_router.add_ocean_token(
+        newTokenAddress, get_factory_deployer_wallet(network=_NETWORK)
+    )
+
+    assert factory_router.ocean_tokens(newTokenAddress) == True
+    factory_router.remove_ocean_token(
+        newTokenAddress, get_factory_deployer_wallet(network=_NETWORK)
+    )
+    assert factory_router.ocean_tokens(newTokenAddress) == False
+
+
+def test_fail_remove_token(web3, config):
+    newTokenAddress = create_new_token(web3, config)
+
+    factory_router = FactoryRouter(web3, get_factory_router_address(config))
+    factory_router.add_ocean_token(
+        newTokenAddress, get_factory_deployer_wallet(network=_NETWORK)
+    )
+    assert factory_router.ocean_tokens(newTokenAddress) == True
+    with pytest.raises(exceptions.ContractLogicError):
+        factory_router.remove_ocean_token(newTokenAddress, get_consumer_wallet())
+
+
+def test_update_opf_fee(web3, config):
+    init(web3, config)
+    newTokenAddress = create_new_token(web3, config)
+    factory_deployer = get_factory_deployer_wallet(network=_NETWORK)
+
+    factory_router = FactoryRouter(web3, get_factory_router_address(config))
+
+    assert factory_router.ocean_tokens(newTokenAddress) == False
+    assert factory_router.get_opf_fee(newTokenAddress) == web3.toWei(0.001, "ether")
+    assert factory_router.swap_ocean_fee() == web3.toWei(0.001, "ether")
+
+    factory_router.update_opf_fee(web3.toWei(0.01, "ether"), factory_deployer)
+    assert factory_router.ocean_tokens(newTokenAddress) == False
+    assert factory_router.get_opf_fee(newTokenAddress) == web3.toWei(0.01, "ether")
+    assert factory_router.swap_ocean_fee() == web3.toWei(0.01, "ether")
+
+
+def test_fail_update_opf_fee(web3, config):
+    init(web3, config)
```

Comment:
Instead of using the init function, write the functionality of it. It's about 2 lines of code. And also the name of the function is not suitable for updating the OPF Fee.